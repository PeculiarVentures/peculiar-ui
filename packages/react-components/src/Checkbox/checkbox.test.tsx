import React from 'react';
import {
  renderWithWrapper as render,
  screen,
  userEvent,
  fireEvent,
} from '../test-utils';
import { Checkbox } from '../index';
// import { PlusIcon } from '../icons';

describe('<Checkbox />', () => {
  describe('Checkbox render variants', () => {
    it('should render as default', () => {
      const { asFragment } = render(<Checkbox id="test-id" />);
      const input = screen.getByRole('checkbox');

      expect(input).toBeInTheDocument();
      expect(input.tagName).toBe('INPUT');
      expect(input.getAttribute('type')).toBe('checkbox');
      expect(input).not.toBeChecked();

      const label = input.closest('label');

      expect(label).toBeInTheDocument();
      expect(label.getAttribute('for')).toEqual(input.id);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have id', () => {
      render(<Checkbox id="test-id" />);

      const input = screen.getByRole('checkbox');

      expect(input.getAttribute('id')).toBe('test-id');

      const label = input.closest('label');

      expect(label).toBeInTheDocument();
      expect(label.getAttribute('for')).toEqual(input.id);
    });

    it('should be disabled', () => {
      const { asFragment } = render(<Checkbox id="test-id" disabled />);
      const input = screen.getByRole('checkbox');

      expect(input).toHaveAttribute('disabled');

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have required', () => {
      const { asFragment } = render(<Checkbox id="test-id" required />);
      const input = screen.getByRole('checkbox');

      expect(input).toHaveAttribute('required');

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have test id', () => {
      const { asFragment } = render(<Checkbox id="test-id" data-testid="test-id" />);
      const input = screen.getByRole('checkbox');
      const label = input.closest('label');

      expect(label.getAttribute('data-testid')).toMatch(/test-id/i);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have className', () => {
      const { asFragment } = render(<Checkbox id="test-id" className="test-cls" />);
      const input = screen.getByRole('checkbox');
      const label = input.closest('label');

      expect(label.getAttribute('class')).toMatch(/test-cls/i);

      expect(asFragment()).toMatchSnapshot();
    });

    // TODO: do not passed, should fix in component
    // it('should have input className', () => {
    //   render(<Checkbox inputProps={{ className: 'test-cls' }} />);

    //   const input = screen.getByRole('checkbox');

    //   expect(input.getAttribute('class')).toMatch(/test-cls/i);
    // });

    // it('should have custom checked icon', () => {
    //   render(<Checkbox checkedIcon={<PlusIcon data-testid="icon" />} />);

    //   const icon = screen.getByTestId('icon');

    //   expect(icon).toBeInTheDocument();
    //   expect(icon.tagName).toBe('svg');
    //   expect(icon.getAttribute('class')).toMatch(/Checkbox-icon/i);
    // });

    it('should be checked, defaultChecked prop', () => {
      const { asFragment } = render(<Checkbox id="test-id" defaultChecked />);
      const input = screen.getByRole('checkbox');

      expect(input).toBeChecked();

      expect(asFragment()).toMatchSnapshot();
    });

    it('should be unchecked, defaultChecked prop', () => {
      render(<Checkbox defaultChecked={false} />);

      const input = screen.getByRole('checkbox');

      expect(input).not.toBeChecked();
    });

    it('should be checked', () => {
      const onChange = jest.fn();
      const { asFragment } = render(
        <Checkbox
          id="test-id"
          checked
          onChange={onChange}
        />,
      );

      expect(asFragment()).toMatchSnapshot();

      const input = screen.getByRole('checkbox');

      expect(input).toBeChecked();
    });

    it('should be unchecked', () => {
      const onChange = jest.fn();

      render(
        <Checkbox
          id="test-id"
          checked={false}
          onChange={onChange}
        />,
      );

      const input = screen.getByRole('checkbox');

      expect(input).not.toBeChecked();
    });

    it('should forwards ref to label element', () => {
      const ref = React.createRef<HTMLLabelElement>();

      render(<Checkbox ref={ref} />);

      const input = screen.getByRole('checkbox');

      expect(input.closest('label')).toEqual(ref.current);
    });
  });

  describe('Checkbox focus behaviour', () => {
    it('should have focus', async () => {
      render(<Checkbox />);

      const input = screen.getByRole('checkbox');

      expect(document.activeElement).not.toEqual(input);
      await userEvent.tab();
      expect(document.activeElement).toEqual(input);
      expect(input).toHaveFocus();
    });

    it("shouldn't have focus when disabled has been passed to the component", async () => {
      render(<Checkbox disabled />);

      const input = screen.getByRole('checkbox');

      expect(document.activeElement).not.toEqual(input);
      await userEvent.tab();
      expect(document.activeElement).not.toEqual(input);
      expect(input).not.toHaveFocus();
    });
  });

  describe('Checkbox click behaviour', () => {
    it('should be checked when clicked', () => {
      render(<Checkbox id="test-id" />);

      const input = screen.getByRole('checkbox');

      expect(input).not.toBeChecked();
      fireEvent.click(input);
      expect(input).toBeChecked();
    });

    it('should be unchecked when clicked', () => {
      render(<Checkbox id="test-id" defaultChecked />);

      const input = screen.getByRole('checkbox');

      expect(input).toBeChecked();
      fireEvent.click(input);
      expect(input).not.toBeChecked();
    });

    it('should calls onClick prop when clicked', () => {
      const handleClick = jest.fn();

      render(<Checkbox id="test-id" onClick={handleClick} />);

      fireEvent.click(screen.getByRole('checkbox'));
      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('should calls onChange prop when clicked', () => {
      const onChange = jest.fn();

      render(<Checkbox id="test-id" onChange={onChange} />);

      const input = screen.getByRole('checkbox');

      expect(onChange).toBeCalledTimes(0);
      fireEvent.click(input);
      expect(onChange).toBeCalledTimes(1);
    });
  });
});
