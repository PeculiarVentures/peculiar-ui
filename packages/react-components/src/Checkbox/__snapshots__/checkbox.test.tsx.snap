// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Checkbox render variants should be checked 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  position: relative;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 2px;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  color: var(--pv-color-gray-9);
  border-color: currentColor;
  background-color: transparent;
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
  background-color: var(--pv-color-primary);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
  background-color: var(--pv-color-gray-6);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  color: var(--pv-color-white);
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      checked=""
      class="emotion-1"
      id="test-id"
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.338 11.656l5.98 5.667a1 1 0 001.452-.081l7.144-8.453a.51.51 0 00-.043-.716l-.38-.338a.508.508 0 00-.717.043l-6.84 7.7-5.584-4.96a.509.509 0 00-.717.042l-.337.38a.51.51 0 00.042.716z"
        fill="currentColor"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Checkbox /> Checkbox render variants should be checked, defaultChecked prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  position: relative;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 2px;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  color: var(--pv-color-gray-9);
  border-color: currentColor;
  background-color: transparent;
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
  background-color: var(--pv-color-primary);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
  background-color: var(--pv-color-gray-6);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  color: var(--pv-color-white);
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      checked=""
      class="emotion-1"
      id="test-id"
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.338 11.656l5.98 5.667a1 1 0 001.452-.081l7.144-8.453a.51.51 0 00-.043-.716l-.38-.338a.508.508 0 00-.717.043l-6.84 7.7-5.584-4.96a.509.509 0 00-.717.042l-.337.38a.51.51 0 00.042.716z"
        fill="currentColor"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Checkbox /> Checkbox render variants should be disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  position: relative;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 2px;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  color: var(--pv-color-gray-9);
  border-color: currentColor;
  background-color: transparent;
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
  background-color: var(--pv-color-primary);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
  background-color: var(--pv-color-gray-6);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  color: var(--pv-color-white);
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      disabled=""
      id="test-id"
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.338 11.656l5.98 5.667a1 1 0 001.452-.081l7.144-8.453a.51.51 0 00-.043-.716l-.38-.338a.508.508 0 00-.717.043l-6.84 7.7-5.584-4.96a.509.509 0 00-.717.042l-.337.38a.51.51 0 00.042.716z"
        fill="currentColor"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Checkbox /> Checkbox render variants should have className 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  position: relative;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 2px;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  color: var(--pv-color-gray-9);
  border-color: currentColor;
  background-color: transparent;
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
  background-color: var(--pv-color-primary);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
  background-color: var(--pv-color-gray-6);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  color: var(--pv-color-white);
  pointer-events: none;
}

<label
    class="test-cls emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.338 11.656l5.98 5.667a1 1 0 001.452-.081l7.144-8.453a.51.51 0 00-.043-.716l-.38-.338a.508.508 0 00-.717.043l-6.84 7.7-5.584-4.96a.509.509 0 00-.717.042l-.337.38a.51.51 0 00.042.716z"
        fill="currentColor"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Checkbox /> Checkbox render variants should have custom checked icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  position: relative;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 2px;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  color: var(--pv-color-gray-9);
  border-color: currentColor;
  background-color: transparent;
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
  background-color: var(--pv-color-primary);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
  background-color: var(--pv-color-gray-6);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  color: var(--pv-color-white);
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 12.5h12M12 18.5v-12"
        stroke="currentColor"
        stroke-width="1.5"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Checkbox /> Checkbox render variants should have required 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  position: relative;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 2px;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  color: var(--pv-color-gray-9);
  border-color: currentColor;
  background-color: transparent;
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
  background-color: var(--pv-color-primary);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
  background-color: var(--pv-color-gray-6);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  color: var(--pv-color-white);
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      required=""
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.338 11.656l5.98 5.667a1 1 0 001.452-.081l7.144-8.453a.51.51 0 00-.043-.716l-.38-.338a.508.508 0 00-.717.043l-6.84 7.7-5.584-4.96a.509.509 0 00-.717.042l-.337.38a.51.51 0 00.042.716z"
        fill="currentColor"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Checkbox /> Checkbox render variants should have test id 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  position: relative;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 2px;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  color: var(--pv-color-gray-9);
  border-color: currentColor;
  background-color: transparent;
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
  background-color: var(--pv-color-primary);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
  background-color: var(--pv-color-gray-6);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  color: var(--pv-color-white);
  pointer-events: none;
}

<label
    class="emotion-0"
    data-testid="test-id"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.338 11.656l5.98 5.667a1 1 0 001.452-.081l7.144-8.453a.51.51 0 00-.043-.716l-.38-.338a.508.508 0 00-.717.043l-6.84 7.7-5.584-4.96a.509.509 0 00-.717.042l-.337.38a.51.51 0 00.042.716z"
        fill="currentColor"
      />
    </svg>
  </label>
</DocumentFragment>
`;

exports[`<Checkbox /> Checkbox render variants should render as default 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: var(--pv-size-base-4);
  height: var(--pv-size-base-4);
  position: relative;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  outline: 0;
  border-radius: 2px;
  border-width: 2px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  color: var(--pv-color-gray-9);
  border-color: currentColor;
  background-color: transparent;
}

.emotion-1:before {
  top: -7px;
  left: -7px;
  right: -7px;
  bottom: -7px;
  content: "";
  position: absolute;
  border-radius: 50%;
  opacity: 0;
  background-color: currentColor;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1:checked+[aria-hidden] {
  opacity: 1;
}

.emotion-1:not(:disabled) {
  cursor: pointer;
}

.emotion-1:not(:disabled):checked {
  color: var(--pv-color-primary-shade-1);
  background-color: var(--pv-color-primary);
}

.emotion-1:not(:disabled):hover:before {
  opacity: 0.18;
}

.emotion-1:not(:disabled):focus:before {
  opacity: 0.23;
}

.emotion-1:not(:disabled):active:before {
  opacity: 0.3;
}

.emotion-1:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-6);
}

.emotion-1:disabled:checked {
  color: var(--pv-color-gray-7);
  background-color: var(--pv-color-gray-6);
}

.emotion-2 {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  color: var(--pv-color-white);
  pointer-events: none;
}

<label
    class="emotion-0"
    for="test-id"
  >
    <input
      class="emotion-1"
      id="test-id"
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="emotion-2"
      fill="none"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.338 11.656l5.98 5.667a1 1 0 001.452-.081l7.144-8.453a.51.51 0 00-.043-.716l-.38-.338a.508.508 0 00-.717.043l-6.84 7.7-5.584-4.96a.509.509 0 00-.717.042l-.337.38a.51.51 0 00.042.716z"
        fill="currentColor"
      />
    </svg>
  </label>
</DocumentFragment>
`;
